ios/CodePush/CodePush.m

```diff
@@ -760,4 +760,29 @@ - (void)applicationWillResignActive
     resolve(nil);
 }
 
+#pragma mark - Methods for handling dispatching of download progress events to JS (Private)
+
+long long latestExpectedContentLength = -1;
+long long latestReceivedConentLength = -1;
+
+- (void)didUpdateFrame:(RCTFrameUpdate *)update
+{
+    // Notify the script-side about the progress
+    [self.bridge.eventDispatcher
+     sendDeviceEventWithName:@"CodePushDownloadProgress"
+     body:@{
+            @"totalBytes":[NSNumber numberWithLongLong:latestExpectedContentLength],
+            @"receivedBytes":[NSNumber numberWithLongLong:latestReceivedConentLength]
+           }];
+    _paused = YES;
+}
+
+- (void)sendDownloadProgressDuringNextFrame:(long long)expectedContentLength
```

Comment:
How about `updateDownloadProgressForNextFrame`? We're not actually sending every download progress event, since they're being throttled.